File: 프로그래머스/Level2/[3차] n진수 게임.cpp
#include <string>
#include <vector>

using namespace std;

string calNum(int num, int n){
    string tmp = "";
    string number = "0123456789ABCDEF";
    
    while(num/n){
        tmp = number[num%n] + tmp;
        num /= n;
    }
    
    tmp = number[num%n] + tmp;
    return tmp;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    string allAnswer = "";
    
    for(int i=0; i<t*m; i++){ 
        allAnswer += calNum(i,n);
    }
    int count = 0;
    for(int i=p-1; i<allAnswer.size(); i+=m){
        answer += allAnswer[i];
        if(count +1 == t) break;
        count ++;
    }
    return answer;
}

File: 프로그래머스/Level2/[3차] n진수 게임.js
function solution(n, t, m, p) {
    let answer = '';
    let tmp = "";
    
    for(let i=0; i<t*m; i++)
        tmp += i.toString(n);

    for(let i=0; i<t*m; i++)
        if(i%m === p-1) answer += tmp[i];

    return answer.toUpperCase();
}
